packages/rest/src/request-context.ts

```diff
@@ -6,17 +6,90 @@
 import {Context} from '@loopback/context';
 import * as onFinished from 'on-finished';
 import {RestBindings} from './keys';
+import {RestServerResolvedConfig} from './rest.server';
 import {HandlerContext, Request, Response} from './types';
 
 /**
  * A per-request Context combining an IoC container with handler context
  * (request, response, etc.).
  */
 export class RequestContext extends Context implements HandlerContext {
+  /**
+   * Get the protocol used by the client to make the request.
+   * Please note this protocol may be different from what we are observing
+   * at HTTP/TCP level, because reverse proxies like nginx or sidecars like
+   * Envoy are switching between protocols.
+   */
+  get requestedProtocol(): string {
```

Comment:
Shouldn't we add unit test for this? We want to be able to detect any unexpected change in future.

Reply:
Good catch. I was being lazy and relying on the existing integration/acceptance-level tests. Will add unit tests.